Main3

CasTeTRU opened the pull request:
Спринт №15

CasTeTRU committed:
Задание по спринту №15

CasTeTRU committed:
Задание по спринту №15

IliaAnK commented on app/src/main/java/com/example/playlistmaker/data/repository/SearchHistoryRepositoryImpl.kt:
```diff
@@ -0,0 +1,51 @@
+import com.google.gson.reflect.TypeToken
+
+class SearchHistoryRepositoryImpl(private val context: Context) : SearchHistoryRepository {
+    private val prefs = context.getSharedPreferences("search_history", Context.MODE_PRIVATE)
```
Рекомендация:
Это зависимость, которую стоит создавать в `Creator` и передавать сюда через конструктор класса вместо `Context`

IliaAnK commented on app/src/main/java/com/example/playlistmaker/data/repository/SettingsRepositoryImpl.kt:
```diff
@@ -0,0 +1,21 @@
+import com.example.playlistmaker.domain.api.SettingsRepository
+
+class SettingsRepositoryImpl(private val context: Context) : SettingsRepository {
+    private val prefs = context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)
```
Рекомендация:
Это зависимость, которую стоит создавать в `Creator` и передавать сюда через конструктор класса вместо `Context`

IliaAnK commented on app/src/main/java/com/example/playlistmaker/data/repository/TrackRepositoryImpl.kt:
```diff
@@ -0,0 +1,32 @@
+    }
+}
+
+fun Track.toDomain(): TrackDomainModel = TrackDomainModel(
```
Рекомендация:
Можно вынести в отдельный файл

IliaAnK commented on app/src/main/java/com/example/playlistmaker/data/dto/ThemeManager.kt:
```diff
@@ -1,4 +1,4 @@
-package com.example.playlistmaker
+package com.example.playlistmaker.data.dto
```
Обязательно:
По всем признакам это репозиторий
Нужно:
1. Перенести в соответствующую директорию
2. Добавить интерфейс
3. Добавить полноценный `domain` слой
4. контекст можно передать предварительно записав его в поле `Creator` из `App`

IliaAnK commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -1,6 +1,7 @@
 package com.example.playlistmaker
 
 import android.app.Application
+import com.example.playlistmaker.data.dto.ThemeManager
```
Обязательно:
Здесь нужно использовать `Interactor`
Выше как раз написал о его необходимости

IliaAnK commented on app/src/main/java/com/example/playlistmaker/presentation/SettingsActivity.kt:
```diff
@@ -1,29 +1,34 @@
-
-        themeManager = ThemeManager(this)
+        settingsInteractor = Creator.provideSettingsInteractor(this)
+        val themeManager = ThemeManager(this)
```
Обязательно:
Здесь аналогично, нужно использовать `Interactor`

CasTeTRU committed:
Задание по спринту №15
1) Отработаны замечания по Interactorу для App, SettingsActivity
2) У меня автоматический поиск работает, через 2000мс происходит запрос по данным введенным в поисковую строку.
3) Плеер также работает, 30 секундные демо играют

CasTeTRU committed:
Задание по спринту №16

CasTeTRU committed:
Merge pull request #17 from CasTeTRU/main5

main5
